packages/vm/tests/api/runTx.spec.ts

```diff
@@ -334,4 +334,49 @@ tape('runTx() -> API return values', async (t) => {
     }
     t.end()
   })
+
+  t.test('simple run, runTx default return values', async (t) => {
+    for (const txType of TRANSACTION_TYPES) {
+      const vm = new VM({ common })
+      const tx = getTransaction(vm._common, txType.type, true)
+
+      const caller = tx.getSenderAddress()
+      const acc = createAccount()
+      await vm.stateManager.putAccount(caller, acc)
+
+      const res = await vm.runTx({ tx })
+
+      t.deepEqual(
+        res.gasUsed,
+        tx.getBaseFee(),
+        `runTx result -> gasUsed -> tx.getBaseFee() (${txType.name})`
+      )
+      t.deepEqual(
+        res.amountSpent,
+        res.gasUsed.mul(tx.gasPrice),
+        `runTx result -> amountSpent -> gasUsed * gasPrice (${txType.name})`
+      )
+      t.deepEqual(
+        res.bloom.bitvector,
+        Buffer.from('00'.repeat(256), 'hex'),
+        `runTx result -> bloom.bitvector -> should be empty (${txType.name})`
+      )
+      t.deepEqual(
+        res.receipt.gasUsed,
+        res.gasUsed.toArrayLike(Buffer),
+        `runTx result -> receipt.gasUsed -> result.gasUsed as Buffer (${txType.name})`
+      )
```

Comment:
This is actually not so beautiful that we here return the `gasUsed` as a `Buffer` and on the upper level as a `BN` value. 🙄  Should we add this to the list of breaking changes for v6 #1024 for an update (to `BN` I guess) or is there some reasoning here I overlook? //cc @ryanio

Reply:
ah yes I think we should update it to return a `BN`.

in general here I'd say that instead of using `t.deepEqual` we should use `BN`'s `eq` method, so for example `t.ok(res.gasUsed.eq(new BN(res.receipt.gasUsed)))`

I also like to do this with `Buffer`s as well: `t.ok(buffer1.equals(buffer2))`

Reply:
Is this encompassing more checks or stricter or what's the reasoning for this?